backend/src/test/java/de/tobias/intestinalinspector/controller/UserControllerTest.java

```diff
@@ -0,0 +1,77 @@
+package de.tobias.intestinalinspector.controller;
+
+import de.tobias.intestinalinspector.TestAuthorization;
+import de.tobias.intestinalinspector.api.UserDto;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(
+        properties = "spring.profiles.active:h2",
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class UserControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String url(){
+        return "http://localhost:"+port+"/api/user";
+    }
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+
+    @Test
+    @Order(1)
```

Comment:
Make your tests independent of each other. A good solution would be to autowire the repo in the test and clear it after each test run.

Reply:
During testing of FoodController and PainController I received following issue:

By using @BeforeEach and @AfterEach I was able to set and clean database entries for every test independent. Nevertheless, the internal indexing of h2 database was not reset. This leads to problems in tests which are based on a specific index, like update( by id) or delete( by id) methods. 

For example: If I use @BeforeEach to generate a database entry, then test the update method and afterwards delete the database entry the index for this entry can have two possible values: If I use the single run of the test the index is 1. Because the database is set up only for this specific test.  
But if I run the test together with all other tests of a controller, the index changes to a higher number. Since several other tests (together with @BeforeEach and @Aftereach methods) ran before the program is reaching my update tests. 
Which means on this point, I had still only one database entry for testing, but the index was not 1 but 5. 
(Because index is not resetting)

My current solution is use of following Annotation:
@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)

This annotation marks ApplicationContext associated with a test as dirty and a new context will be set. Therefore a new indexing starting by 1 for each test. 
Pro: Tests are running now independent in single use as well as together
Con: Testing is slowed down by the addional generating of new context each test.

Reply:
I would not initialize the repo in a @BeforeEach, but in every single test with a specific setup for that test. When adding entities to the db you get an entity with the id back,that you can use later for your assertion - no problem there.